lib/charms/layer/jenkins/plugins.py

```diff
@@ -8,9 +9,27 @@
 from jenkins_plugin_manager.plugin import UpdateCenter
 
 
+class PluginSiteError(Exception):
+    def __init__(self):
+        self.message = ("The configured plugin-site doesn't provide an "
+                        "update-center.json file or is not acessible.")
+
+
 class Plugins(object):
     """Manage Jenkins plugins."""
 
+    def __init__(self):
+        if hookenv.config()["plugins-site"] == "https://updates.jenkins-ci.org/latest/":
+            self.update_center = UpdateCenter()
+        else:
+            plugins_site = hookenv.config()["plugins-site"]
+            try:
+                update_center = plugins_site + "/update-center.json"
+                urllib.request.urlopen(update_center)
+                self.update_center = UpdateCenter(uc_url=update_center)
+            except urllib.error.HTTPError:
+                raise PluginSiteError()
```

Comment:
I think it's worth having a message in here with a bit more detail about the failure.